### DISCUSSION TITLE
NuTone non-Plus switch/dimmers send ApplicationUpdateRequest on double-tap. Can we make it a triggerable event?

### DISCUSSION THREAD

**Discussion by nmpu** [zwave-js/zwave-js#6643] [Q&A] [answered]
I discovered that my non-Plus **NuTone** switches and dimmers will send `ApplicationUpdateRequest` on a double-tap. This must have been an undocumented feature for engineering or tech support. I'd like to make use of it to cycle through presets on my dimmers. It would also be useful on the switches. It's somewhat limited in that you can't distinguish between up and down and it also affects the load, It's only useful if you double-tap up while on or double-tap down while off.

The **Z-Wave JS** response is to send `MultilevelSwitchCCGet`. That's perfectly reasonable. Since I can't trigger directly on the `ApplicationUpdateRequest`, I watch for the `MultilevelSwitchCCReport` which will never arrive unsolicited. The only other time I see `MultilevelSwitchCCReport` is after `MultilevelSwitchCCSet` which I can anticipate. I already have all my dimmers templated due to [HA rounding errors](https://community.home-assistant.io/t/conversion-between-brightness-pct-1-100-z-wave-1-99-and-brightness-1-255/661523), so I know when an update will be sent.  I'm pretty happy with what I've got now, but I have to use a 5-second guard time to ensure I'm seeing the double-tap and not some remnant of a set command. This is because the dimmer step parameters are not configurable and it can take a long time to ramp before the final report is sent. I also have [this problem](https://github.com/zwave-js/node-zwave-js/discussions/6641).

Here are my questions:

1) I would like to trigger **HA** directly off the `ApplicationUpdateRequest`. Is this too esoteric to ever be considered as a feature request?
2) If I were to make the changes myself, is this something that could be included in the mainstream code?

**Comment by AlCalzone** [comment]
`ApplicationUpdateRequest` was used by older devices to get around a patent related to instant status updates. The same used to be done using `Hail CC` (not sure which one came first). So essentially, receiving this command means there was something changed on the node. Are you getting the same when single-pressing?

Some other devices send a BasicCCSet in this case, which gets exposed as a "Basic event" value. We could probably hijack this, if double-tap is truly the only situation where your devices send that command. You won't get the info whether it was a double tap up or down though.

**Comment by nmpu** [comment]
These particular **NuTone** devices are totally silent unless spoken to. The devices don't send anything for on/off or ramp. The instruction sheet says nothing about association groups. A single-tap works for inclusion, but a double-tap is required for exclusion. I've seen multiple posts where people confuse the double-tap activity:

[[question] Device only update status after tapping (off) twice (manual)](https://github.com/zwave-js/node-zwave-js/issues/4220)
[Zwave Device not allowing associations](https://github.com/zwave-js/node-zwave-js/issues/4864)

Can you work some magic to make this discussion a feature request?

**Comment by AlCalzone** [comment, marked as answer]
Feature request here: https://github.com/zwave-js/node-zwave-js/issues/6655
